tasks/actorstate/miner/sector_events.go

```diff
@@ -54,37 +55,42 @@ func (SectorEventsExtractor) Extract(ctx context.Context, a actorstate.ActorInfo
 
 	} else {
 		// If the miner has previous state compute the list of new sectors and precommit in its current state.
-		preCommitChanges, err = node.DiffPreCommits(ctx, a.Address, a.Current, a.Executed, ec.PrevState, ec.CurrState)
+		preCommitChanges, err = node.DiffPreCommits(ctx, a.Address, a.Current, a.Executed, ec.ParentState(), ec.CurrentState())
 		if err != nil {
 			return nil, err
 		}
 
-		sectorChanges, err = node.DiffSectors(ctx, a.Address, a.Current, a.Executed, ec.PrevState, ec.CurrState)
+		sectorChanges, err = node.DiffSectors(ctx, a.Address, a.Current, a.Executed, ec.ParentState(), ec.CurrentState())
 		if err != nil {
 			return nil, err
 		}
 	}
 
-	sectorEventModel, err := extractSectorEvents(ctx, a, ec, sectorChanges, preCommitChanges)
+	dlDiff, err := miner.DiffDeadlines(ec.ParentState(), ec.CurrentState())
```

Comment:
moved to top-level method to improve testability of the different ExtractSector* methods.